screens/Meal.tsx

```diff
@@ -1,210 +1,303 @@
+  Box,
+  Center,
+  HStack,
+  ScrollView,
+  Text,
+  VStack,
+  Modal,
+  Button,
+} from 'native-base';
+import _, {size} from 'lodash';
+import axios from 'axios';
+import {parse} from 'node-html-parser';
 
 type Props = BottomTabScreenProps<RootTabList, 'Meal'>;
 
 export default function Meal({navigation}: Props) {
-  // todo
-  //   const favoritedNumbers = DUMMY_MEAL_INFO.map(({isFavorited}) =>
-  //     Number(isFavorited),
-  //   ).reduce((sum, currValue) => sum + currValue);
-  //   const notFavoritedNumbers = DUMMY_MEAL_INFO.length - favoritedNumbers;
+  type TodaysMenu = {
+    [name: string]: {
+      breakfast: string;
+      lunch: string;
+      dinner: string;
+      contact: string;
+    };
+  };
+  // state 선언
+  const [menu, setMenu] = useState<TodaysMenu | null>(null); // store menu data here
+  const [isFavorite, setIsFavorite] = useState<{} | null>(null); // store favorite or not here
+  const [selectedMeal, setSelectedMeal] = useState<string | null>(null); // store selected meal (modal) here
+
+  // 식단 정보 크롤링
+  const exampleDateURL =
+    'https://snuco.snu.ac.kr/ko/foodmenu?field_menu_date_value_1%5Bvalue%5D%5Bdate%5D=&field_menu_date_value%5Bvalue%5D%5Bdate%5D=09%2F23%2F2021';
+  const todayMenuURL = 'https://snuco.snu.ac.kr/ko/foodmenu'; // 정보 받아와서 리스트에 저장, 각 식당의 리스트 인덱스 따로 저장
+
+  useEffect(() => {
+    axios.get(exampleDateURL).then(res => {
+      const html = res.data;
+      const root = parse(html);
+      const data: TodaysMenu = {};
+      _.chain(root.querySelector('tbody').childNodes)
+        .map(trNode => {
+          const trTexts = _.chain(trNode.childNodes)
+            .map((tdNode, idx) =>
+              tdNode.innerText
+                .split(/\s|\t|\n/)
+                .filter(item => item.length > 0)
+                .join(' '),
+            )
+            .value();
+          return trTexts;
+        })
+        .filter(trTexts => trTexts.length > 0)
+        .forEach((trTexts, idx) => {
+          const [a, nameAndContact, c, breakfast, e, lunch, g, dinner] =
+            trTexts;
+          const splitedNameAndContact = _.split(nameAndContact, '(');
+          const name = splitedNameAndContact[0];
+          const contact = splitedNameAndContact[1].slice(0, -1);
+          data[name] = {breakfast, lunch, dinner, contact};
+        })
+        .value();
+      setMenu(data);
+    });
+  }, []);
+
+  // 식당 일반 운영정보 크롤링
+
+  // 즐찾, 운영중에 맞게 세팅
+  useEffect(() => {
+    async function setFavoriteStates() {
+      const tempData = await Promise.all(
+        DUMMY_MEAL_INFO.map(async meal => {
+          const trueOrFalse = await getIsFavorite(meal.name);
+          return {[meal.name]: trueOrFalse};
+        }),
+      );
+      const data = Object.assign({}, ...tempData);
+      setIsFavorite(data);
+    }
+    setFavoriteStates();
+  }, []);
+
+  console.log(isFavorite);
+
+  const storeIsFavorite = async (mealName: string | null) => {
+    if (mealName === null) {
+      return;
+    }
```

Comment:
로직의 복잡도를 낮추기 위해 null이 들어오는 상황을 대비하는 것 보다 null이 아예 들어오지 않게 하는 것이 좋을듯
```suggestion
  const storeIsFavorite = async (mealName: string) => {
```

Reply:
코드 새로 정리하면서 반영하겠음